### PR TITLE
Add commands to remove advertised routes of tailscale in k3s-killall.sh

### PR DESCRIPTION
**Pull request by manuelbuil**
#### Proposed Changes ####

When k3s gets removed from the node, the tailscale interface should remove the routes that it had advertised: 
```
tailscale set --advertise-routes=
```
This PR adds those commands to k3s-killall.sh

#### Types of Changes ####

Bugfix

#### Verification ####

After running k3s-killall.sh, the tailscale interface should not have any IP and running `tailscale status --json` should not report anything because we logged out

#### Linked Issues ####

https://github.com/k3s-io/k3s/issues/7772

#### User-Facing Change ####

```release-note

```

### CODE REVIEW

**Review by brandond — COMMENTED** [review]
I'm a little concerned that this will be a breaking change for anyone using tailscale standalone. I know from GH issue that there are users that are currently deploying tailscale by hand, and then installing k3s afterwards. Is there anything we can do to avoid breaking things? Is there some way to uniquely configure tailscale to indicate that it is managed by k3s, and check for that when uninstalling?

**Comment by manuelbuil** [comment]
> I'm a little concerned that this will be a breaking change for anyone using tailscale standalone. I know from GH issues that there are users that are currently deploying tailscale by hand, and then installing k3s afterwards. Is there anything we can do to avoid breaking things? Is there some way to uniquely configure tailscale to indicate that it is managed by k3s, and check for that when uninstalling?

Ok, I did not think about this use case TBH. Instead of turning it off, we could remove the configured subnet route instead, which is what we basically do apart from starting and logging. That would still fix the issue. Let me update the PR

**Review by dereknola — APPROVED** [review]

File: install.sh [hunk 1]
```diff
@@ -739,6 +739,27 @@ killtree() {
...
+    # Delete cni related interfaces
+    ip link delete cni0
+    ip link delete flannel.1
+    ip link delete flannel-v6.1
+    ip link delete kube-ipvs0
+    ip link delete flannel-wg
+    ip link delete flannel-wg-v6
+
+    # Restart tailscale
+    if [[ -n $(command -v tailscale) ]]; then
```
Comment (brandond) [inline install.sh:758]: Single braces only, this isn't bash. And quote the potentially empty subshell output.

```suggestion
    if [ -n "$(command -v tailscale)" ]; then
```

**Review by brandond — CHANGES_REQUESTED** [review]

**Review by brandond — APPROVED** [review]

**Comment by DerRockWolf** [comment]
> I'm a little concerned that this will be a breaking change for anyone using tailscale standalone. I know from GH issues that there are users that are currently deploying tailscale by hand, and then installing k3s afterwards. Is there anything we can do to avoid breaking things? Is there some way to uniquely configure tailscale to indicate that it is managed by k3s, and check for that when uninstalling?

Removing the advertised routes is also somewhat breaking, if one is using manually created advertised routes.

Is the `k3s-killall.sh` script being executing automatically at some point, or is it always run manually?
If it's only run manually I'm fine with it removing the advertised routes.

**Comment by brandond** [comment]
It is never run automatically. An administrator has to chose to run the uninstall or killall script, although a killall is also run as part of the uninstall (for obvious reasons).

**Comment by manuelbuil** [comment]
We have this related issue in the backlog: https://github.com/k3s-io/k3s/issues/8012

**Pull request merged**